File: CoWorkHub/CoWorkHub.Model/City.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class City
    {
        public int CityId { get; set; }
        public string CityName { get; set; } = null!;
        public string PostalCode { get; set; } = null!;
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public int CountryId { get; set; }
        public virtual Country Country { get; set; } = null!;
        public bool IsDeleted { get; set; } = false;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/city.dart
import 'package:coworkhub_desktop/models/country.dart';
import 'package:json_annotation/json_annotation.dart';
part 'city.g.dart';

@JsonSerializable()
class City {
  final int cityId;
  final String cityName;
  final String postalCode;
  final double? latitude;
  final double? longitude;
  final int countryId;
  final Country country;

  City({
    required this.cityId,
    required this.cityName,
    required this.postalCode,
    this.latitude,
    this.longitude,
    required this.countryId,
    required this.country,
  });

  factory City.fromJson(Map<String, dynamic> json) {
    return City(
      cityId: json['cityId'],
      cityName: json['cityName'],
      postalCode: json['postalCode'],
      latitude: json['latitude']?.toDouble(),
      longitude: json['longitude']?.toDouble(),
      countryId: json['countryId'],
      country: json['country'] != null
          ? Country.fromJson(json['country'])
          : Country(countryId: json['countryId'] ?? 0, countryName: ""),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'cityId': cityId,
      'cityName': cityName,
      'postalCode': postalCode,
      'latitude': latitude,
      'longitude': longitude,
      'countryId': countryId,
      'country': country.toJson(),
    };
  }
}
